src/algebra/module/localized_module.lean

```diff
@@ -965,6 +971,215 @@ begin
   exact ⟨⟨m, s⟩, mk'_eq_iff.mpr e.symm⟩
 end
 
+@[simp]
+lemma iso_mk' (m : M) (s : S) :
+  (is_localized_module.iso S f).symm (is_localized_module.mk' f m s) = localized_module.mk m s :=
+(is_localized_module.iso S f).symm_apply_apply (localized_module.mk m s)
+
+@[simp]
+lemma lift_mk' (g : M →ₗ[R] M'')
+  (h : ∀ (x : S), is_unit ((algebra_map R (module.End R M'')) x)) (m : M) (s : S) :
+  (is_localized_module.lift S f g h) (is_localized_module.mk' f m s) = (h s).unit⁻¹ (g m) :=
+by { delta is_localized_module.lift, dsimp,
+  rw [is_localized_module.iso_mk', localized_module.lift_mk], refl }
+
 end is_localized_module
 
+section submodule
+
+variable (N : submodule R M)
+
+/-- The localization of a submodule of `M` in `localized_module S M`. -/
+def submodule.localize : submodule (localization S) (localized_module S M) :=
+{ carrier := { x | ∃ (y ∈ N) s, localized_module.mk y s = x },
+  add_mem' := begin
+    rintros _ _ ⟨x, hx, s, rfl⟩ ⟨x', hx', s', rfl⟩,
+    refine ⟨_, _, _, localized_module.mk_add_mk⟩,
+    apply add_mem; apply submodule.smul_mem; assumption
+  end,
+  zero_mem' := ⟨_, zero_mem _, _, rfl⟩,
+  smul_mem' := begin
+    rintros r _ ⟨x, hx, s, rfl⟩,
+    induction r using localization.induction_on,
+    exact ⟨_, N.smul_mem _ hx, _, rfl⟩,
+  end }
```

Comment:
I was thinking if there isn't a neater way to define this, using something like [`submodule.map₂`](https://leanprover-community.github.io/mathlib_docs/algebra/module/submodule/bilinear.html#submodule.map%E2%82%82). I can't think of anything at the moment but you might have a better idea.